concurrency-primer.tex

```diff
@@ -747,6 +747,209 @@ \section{Do we always need sequentially consistent operations?}
 
 \section{Memory orderings}
 
+\subsection{Memory consistency models}
+
+When a program is compiled and executed, it doesn't always follow the written order.
+The system may change the sequence and optimize it to simulate line-by-line execution, as long as the final result matches the expected outcome.
```

Comment:
You should address the impact from modern microprocessors and optimizing compilers.